src/user/user.database.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@nestjs/common'
+import { InjectModel } from '@nestjs/mongoose'
+import { Model } from 'mongoose'
+
+import { USER_MODEL } from '../common/constants/database.constants'
+import { NotFoundError } from '../common/errors/errors'
+import { CreateUserDto } from './dto/create-user.dto'
+import { UpdateUserDto } from './dto/update-user.dto'
+import type { User } from './schemas/user.schema'
+
+@Injectable()
+export class UserDatabaseService {
+  constructor(@InjectModel(USER_MODEL) private userModel: Model<User>) {}
+
+  async getIsUserExist(username: string): Promise<boolean> {
```

Comment:
```suggestion
  async isUserExist(username: string): Promise<boolean> {
```
Also, it's better placing it to the UserService as this check is clearly business-logic.

Reply:
Fixed